### DISCUSSION TITLE
Keycloak and user type

### DISCUSSION THREAD

**Discussion by codesmaker** [Lissy93/dashy#1592] [Help and Support]
Hi,

First of all, thank you a lot for the nice and useful app.

I'm using Dashy with Keycloak which uses Windows AD as the backend authentication system. I'd like to know how can I set the user type in this case. The idea is that I don't want normal users to modify their dashboards.

I know that with the built-in auth, there is an attribute called 'type' whose value can be 'admin' or 'normal' but I couldn't find a similar thing for Keycloak when it's integrated with Windows AD.

Also, there is a question about it but unfortunately, it was not answered. Here it is [https://github.com/Lissy93/dashy/discussions/634](url)

Thanks a lot for all the help/thoughts you can provide.

**Comment by abhi-menasinakai** [comment]
Guys, Even I am facing same issue. I want to disable config settings for the regular users and enable configs for admin. I tried adding client/realm roles to admin user in keycloak. But no use post login. The changes are not reflecting in dashy when I logged in using Keycloak SSO. Can any one know how to modify conf.yaml so that I will be able to get admin and non admin privileges. Official documentation help is not working.!!
